File: backend/src/models/Food.ts
import { BaseEntity, Column, Entity, PrimaryGeneratedColumn } from "typeorm";

export enum FoodType {
    drink = 'Bevanda',
    snack = 'Merenda',
    sandwich = 'Panino'
}

@Entity()
export default class Food extends BaseEntity {
    @PrimaryGeneratedColumn('increment')
    public id!: number;
    
    @Column({type: 'varchar', length: 50, nullable: false})
    public name!: string;

    @Column({type: 'enum', enum: FoodType, nullable: false})
    public type!: FoodType;

    @Column({type: 'int', default: 0})
    public qtyAvaible!: number;

    @Column({type: 'varchar', length: 200, default: ''})
    public description!: string;
}

File: frontend/lib/models/Food.dart
class FoodModel {
  final int id;
  final String name;
  final FoodType type;
  final int quantityAvable;
  final String? description;
  final String imageUrl;
  final double price;

  FoodModel({
    required this.id,
    required this.name,
    required this.quantityAvable,
    required this.imageUrl,
    required this.price,
    this.type = FoodType.sandwich,
    this.description
  });

  factory FoodModel.fromJSON(Map<String,dynamic> data) {
    return FoodModel(
      id: data['id'] as int,
      name: data['name'] as String,
      quantityAvable: data['qtyAvaible'] as int,
      imageUrl: data['imageUrl'] as String,
      price: data['price'].toDouble() ?? 0.0,
    );
  }

  @override
  String toString() => 'FoodModel(id: $id)';
}

enum FoodType {
  drink,
  snack,
  sandwich
}